aiida_common_workflows/workflows/relax/orca/generator.py

```diff
@@ -0,0 +1,141 @@
+        previous_workchain=None,
+        **kwargs
+    ) -> engine.ProcessBuilder:
+        """Return a process builder for the corresponding workchain class with inputs set according to the protocol.
+        :param structure: the structure to be relaxed.
+        :param calc_engines: a dictionary containing the computational resources for the relaxation.
+        :param protocol: the protocol to use when determining the workchain inputs.
+        :param relax_type: the type of relaxation to perform.
+        :param electronic_type: the electronic character that is to be used for the structure.
+        :param spin_type: the spin polarization type to use for the calculation.
+        :param magnetization_per_site: a list with the initial spin polarization for each site. Float or integer in
+            units of electrons. If not defined, the builder will automatically define the initial magnetization if and
+            only if `spin_type != SpinType.NONE`.
+        :param threshold_forces: target threshold for the forces in eV/Å.
+        :param threshold_stress: target threshold for the stress in eV/Å^3.
+        :param previous_workchain: a <Code>RelaxWorkChain node.
+        :param kwargs: any inputs that are specific to the plugin.
+        :return: a `aiida.engine.processes.ProcessBuilder` instance ready to be submitted.
+        """
+        # pylint: disable=too-many-locals
+        protocol = protocol or self.get_default_protocol_name()
+
+        super().get_builder(
+            structure,
+            calc_engines,
+            protocol=protocol,
+            relax_type=relax_type,
+            electronic_type=electronic_type,
+            spin_type=spin_type,
+            magnetization_per_site=magnetization_per_site,
+            threshold_forces=threshold_forces,
+            threshold_stress=threshold_stress,
+            previous_workchain=previous_workchain,
+            **kwargs
+        )
+
+        # Checks
+        if protocol not in self.get_protocol_names():
+            import warnings
+            warnings.warn('no protocol implemented with name {}, using default moderate'.format(protocol))
+            protocol = self.get_default_protocol_name()
+        if 'relax' not in calc_engines:
+            raise ValueError('The `calc_engines` dictionaly must contain "relaxation" as outermost key')
+
+        if magnetization_per_site is not None:
+            print('Warning: magnetization_per_site not supported, ignoring it.')
+
+        if relax_type == RelaxType.NONE:
+            params = params['input_keywords'].remove('Opt')
+
+        if spin_type == SpinType.COLLINEAR:
+            params = params['input_keywords'].append('UKS')
+
+        options = calc_engines['relax']['options']
+
+        params = self._get_params(protocol)
+        params['input_blocks']['pal'].update({
+            'nproc':
+            options['resources']['num_machines'] * options['resources']['num_mpiprocs_per_machine']
```

Comment:
The `num_mpiprocs_per_machine` and `num_machines` are not mandatory in  `resources`. Some logic is needed for situations when they are not specified